libraries/shared/fetcher/log_fetcher.go

```diff
@@ -26,6 +26,7 @@ import (
 
 type ILogFetcher interface {
 	FetchLogs(contractAddresses []common.Address, topics []common.Hash, missingHeader core.Header) ([]types.Log, error)
+	// TODO Extend FetchLogs for doing several blocks at a time
```

Comment:
Yeah - I think there's still some quirkiness if we want to guarantee that we looked up logs associated with each individual header's _hash_ (I think fetch for multiple blocks is only available by number rn), but this comment felt more appropriate here rather than where it used to be